test/language/expressions/object/identifier-shorthand-implements-invalid-strict-mode.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2020 Rick Waldron. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+  Object literal shorthands are only valid with identifier references,
+  not property names. (implements)
+esid: sec-object-initializer
+flags: [noStrict]
+info: |
+  PropertyDefinition:
+    IdentifierReference
+    CoverInitializedName
+    PropertyName : AssignmentExpression
+    MethodDefinition
```

Comment:
Adding this would be pretty helpful:

```
  Identifier : IdentifierName but not ReservedWord
    It is a Syntax Error if this phrase is contained in strict mode code and
    the StringValue of IdentifierName is: "implements", "interface", "let",
    "package", "private", "protected", "public", "static", or "yield". 
```